StrongInject.Tests.Unit/OwnedInjectionTests.cs

```diff
@@ -0,0 +1,1275 @@
+                {
+                    await ((global::System.IAsyncDisposable)b_1_0).DisposeAsync();
+                    throw;
+                }
+
+                return new global::StrongInject.AsyncOwned<global::B>(b_1_0, async () =>
+                {
+                    await ((global::System.IAsyncDisposable)b_1_0).DisposeAsync();
+                });
+            }
+
+            asyncOwned_1_0 = await CreateAsyncOwnedB_1();
+            return asyncOwned_1_0;
+        };
+        a_0_0 = new global::A(b: func_0_1);
+        TResult result;
+        try
+        {
+            result = await func(a_0_0, param);
+        }
+        finally
+        {
+        }
+
+        return result;
+    }
+
+    async global::System.Threading.Tasks.ValueTask<global::StrongInject.AsyncOwned<global::A>> global::StrongInject.IAsyncContainer<global::A>.ResolveAsync()
+    {
+        if (Disposed)
+            throw new global::System.ObjectDisposedException(nameof(Container));
+        global::System.Func<global::System.Threading.Tasks.Task<global::StrongInject.AsyncOwned<global::B>>> func_0_1;
+        global::A a_0_0;
+        func_0_1 = async () =>
+        {
+            global::StrongInject.AsyncOwned<global::B> asyncOwned_1_0;
+            async global::System.Threading.Tasks.Task<global::StrongInject.AsyncOwned<global::B>> CreateAsyncOwnedB_1()
+            {
+                global::B b_1_0;
+                global::System.Threading.Tasks.ValueTask b_1_1;
+                var hasAwaitStarted_b_1_1 = false;
+                b_1_0 = new global::B();
+                try
+                {
+                    b_1_1 = ((global::StrongInject.IRequiresAsyncInitialization)b_1_0).InitializeAsync();
+                    try
+                    {
+                        hasAwaitStarted_b_1_1 = true;
+                        await b_1_1;
+                    }
+                    catch
+                    {
+                        if (!hasAwaitStarted_b_1_1)
+                        {
+                            _ = b_1_1.AsTask().ContinueWith(failedTask => _ = failedTask.Exception, global::System.Threading.Tasks.TaskContinuationOptions.OnlyOnFaulted);
+                        }
+
+                        throw;
+                    }
+                }
+                catch
+                {
+                    await ((global::System.IAsyncDisposable)b_1_0).DisposeAsync();
+                    throw;
+                }
+
+                return new global::StrongInject.AsyncOwned<global::B>(b_1_0, async () =>
+                {
+                    await ((global::System.IAsyncDisposable)b_1_0).DisposeAsync();
+                });
+            }
+
+            asyncOwned_1_0 = await CreateAsyncOwnedB_1();
+            return asyncOwned_1_0;
+        };
+        a_0_0 = new global::A(b: func_0_1);
+        return new global::StrongInject.AsyncOwned<global::A>(a_0_0, async () =>
+        {
+        });
+    }
+}");
+        }
```

Comment:
Tests all look really good.

Only comments:
1. Please add tests for resolving multiple identical async owneds - I think should just work without any problems.
2. The only case which requires an `IAsyncContainer` is when you resolve an `Owned`/`AsyncOwned` with dependencies which require async initialization which isn't inside a func. Note that inside a func, `IAsyncContainer` is irrelevant - all that matters is whether the func returns a `Task`. Also note that `Owned`/`AsyncOwned` is irrelevant for this purpose, since the user is responsible for disposing them not the container.

Reply:
Also it's great to have somebody different write tests because they write tests in a different style, testing aspects I never have!